internal/rest/receipt/receiptstore.go

```diff
@@ -92,11 +83,17 @@ func NewReceiptStore(config *conf.RESTGatewayConf) ReceiptStore {
 	}
 }
 func (r *receiptStore) ValidateConf() error {
-	return r.persistence.validateConf()
+	return r.persistence.ValidateConf()
 }
 
-func (r *receiptStore) Init() error {
-	return r.persistence.Init()
+func (r *receiptStore) Init(ws ws.WebSocketChannels, persistence api.ReceiptStorePersistence) error {
+	r.ws = ws
+	if persistence != nil {
+		r.persistence = persistence
+		return nil
+	} else {
+		return r.persistence.Init()
```

Comment:
Trying to understand this - maybe a comment would be useful?

It seems like the caller of this asserts that `r.persistence` is **not** `nil`, by passing `nil` in for the `persistence` parameter.

Reply:
added inline comments for clarifying these branches (only for testing purposes)